### PR TITLE
[CLI] implement regex filtering

### PR DESCRIPTION
**Pull request by KeenS**
Implement #1911 
To hold backward compatibility, added new fields `*_table_regexes`.

### CODE REVIEW

**Comment by KeenS** [comment]
blocked by #1889 , i guess.

**Comment by weiznich** [comment]
> blocked by #1889 , i guess.

Yes I would like to merge #1889 before merging any bigger PR. (Also maybe releasing 1.4)

**Comment by KeenS** [comment]
Some jobs of CI on Azure is failing with unknown reason and some are canceled somehow.

**Comment by KeenS** [comment]
force rerun resolved everything.

**Comment by KeenS** [comment]
@weiznich It's ready to review.

File: diesel_cli/src/cli.rs [hunk 1]
```diff
@@ -124,32 +124,60 @@ pub fn build_cli() -> App<'static, 'static> {
                 .short("o")
                 .long("only-tables")
                 .help("Only include tables from table-name")
+                .conflicts_with("only-table-regexes")
```
Comment (weiznich) [inline diesel_cli/src/cli.rs]: The next release will be 2.0 so we can make a breaking change here. Therefore I would prefer to just introduce this functionality under the old flags.

cc @diesel-rs/core  as information.

File: diesel_cli/src/main.rs [hunk 2]
```diff
@@ -368,6 +371,12 @@ fn run_infer_schema(matches: &ArgMatches) -> Result<(), Box<Error>> {
         })
         .collect();
 
+    let filter_regex = matches
+        .values_of("table-name-regexpes")
+        .unwrap_or_default()
+        .map(|table_name_regex| Regex::new(table_name_regex).unwrap().into())
```
Comment (weiznich) [inline diesel_cli/src/main.rs]: Could we print a nicer error message here if the provided string is no valid regex?

(Hint: If that map returns a `Result` it is possible to collect into a `Result<Vec<_>, _>`)

**Comment by KeenS** [comment]
fixed message. And also, I found the filtering is referencing non-existent argument. Thus fixed to refer `"table-name"`

```
$ cargo run -p diesel_cli --features postgres --no-default-features -- print-schema --except-table-regexes  'a('
   Compiling diesel_cli v1.4.0 (/home/shun/Rust/diesel/diesel_cli)
    Finished dev [unoptimized + debuginfo] target(s) in 4.28s
     Running `/home/shun/Rust/diesel/target/debug/diesel print-schema --except-table-regexes 'a('`
invalid argument for table filtering regex: regex parse error:
    a(
     ^
error: unclosed group
```

**Comment by KeenS** [comment]
Nightly tests are failing because try_from has been stabilized. It's not relevant to this PR

**Comment by KeenS** [comment]
wait on #2108

**Comment by AlterionX** [comment]
@weiznich @KeenS So is this one good to go and is just waiting for the next major version release? If so, would it be prudent to add in a label or a milestone/project for that? Or is functionality going to be moved back under the old flags?

File: diesel_cli/src/cli.rs (hunk 1 above)
Comment (KeenS) [reply diesel_cli/src/cli.rs]: What does the old flags mean?

Comment (weiznich) [reply diesel_cli/src/cli.rs]: I'm thinking of just replacing the old flag (`only-tables`) with the regex filtering.

**Comment by KeenS** [comment]
> error: 'rustc' is not installed for the toolchain 'beta-x86_64-unknown-linux-gnu'
> To install, run `rustup component add rustc --toolchain beta-x86_64-unknown-linux-gnu`

Maybe travis isn't up-to-date?

**Comment by KeenS** [comment]
closing in favor of #2515. Thank you @TaKO8Ki for taking over my work.

**Pull request closed without merging**